ui/admin/app/controllers/scopes/scope/auth-methods/auth-method/accounts.js

```diff
@@ -0,0 +1,77 @@
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { loading } from 'ember-loading';
+import { confirm } from 'core/decorators/confirm';
+import { notifySuccess, notifyError } from 'core/decorators/notify';
+
+export default class ScopesScopeAuthMethodsAuthMethodAccountsController extends Controller {
+  // =services
+
+  @service can;
+  @service router;
+
+  // =actions
+
+  /**
+   * Rollback changes on an account.
+   * @param {AccountModel} account
+   */
+  @action
+  cancel(account) {
+    const { isNew } = account;
+    account.rollbackAttributes();
+    if (isNew) {
+      this.router.transitionTo(
+        'scopes.scope.auth-methods.auth-method.accounts',
+      );
+    }
+  }
+
+  /**
+   * Save an account in current scope.
+   * @param {AccountModel} account
+   */
+  @action
+  @loading
+  @notifyError(({ message }) => message)
+  @notifySuccess(({ isNew }) =>
+    isNew ? 'notifications.create-success' : 'notifications.save-success',
+  )
+  async save(account, password) {
+    const { isNew } = account;
+    const adapterOptions = {};
+    if (isNew) {
+      adapterOptions.password = password;
+    }
+    await account.save({ adapterOptions });
+    if (this.can.can('read model', account)) {
+      await this.router.transitionTo(
+        'scopes.scope.auth-methods.auth-method.accounts.account',
+        account,
+      );
+    } else {
+      await this.router.transitionTo(
+        'scopes.scope.auth-methods.auth-method.accounts',
+      );
+    }
+    this.router.refresh();
+  }
```

Comment:
I'm noticing that the existing actions (for the most part) use await for transitionTo() and replaceWith(). Though not in all cases is `await` needed, which is confusing.

Reply:
@lisbet-alvarez 
That's a very good point. I wonder why we are using `await` for `transitionTo`. I am checking documentation of the service [here](https://api.emberjs.com/ember/release/classes/RouterService/methods/transitionTo?anchor=transitionTo) and  redirects [documentation here](https://guides.emberjs.com/release/routing/redirection/). None of them talk about anything being returned or the necessity to `await`.

I will suggest to not use `await` and bring this topic to tech time, just in case other team members have a say about it.

Reply:
I think that's a mistake if we are, I don't think it should need to be `await`ed

Reply:
Wait I lied, I just looked up the documentation and the [transition](https://api.emberjs.com/ember/4.12/classes/Transition) is actually `thennable` which makes it in practice a promise. These actually do need to be awaited

Reply:
Nice digging @ZedLi 

After checking the service documentation and read this:
`returns Transition the transition object associated with this attempted transition` and understood `the transtion` object as a plain object, therefore no need for await. But after reading the documentation about transition that you suggested definetly need the `await` for the `thennable`.